src/System.Data.Odbc/src/System/Data/Odbc/OdbcConnection.cs

```diff
@@ -561,7 +562,14 @@ internal void HandleError(OdbcHandle hrHandle, ODBC32.RetCode retcode)
 
         public override void Open()
         {
-            InnerConnection.OpenConnection(this, ConnectionFactory);
+            try
+            {
+                InnerConnection.OpenConnection(this, ConnectionFactory);
+            }
+            catch (DllNotFoundException e) when (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows) && e.Message.Contains(ExternDll.Odbc32))
```

Comment:
It's fragile to grep the text of the exception, even for a string that shouldn't be localized. Messages change: it's not considered a breaking change to do so. I wouldn't bother. Just catch DllNotFoundException

Reply:
(when not on Windows, as you have done)